### PR TITLE
initial ci changes

### PR DESCRIPTION
**Pull request by chayim**
moved gpu back to a docker for testing

valgrind changes to limit to personal branches

reduce valgrind runs on master and integ

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -29,30 +37,30 @@ commands:
       - run:
           name: Setup executor
           command: |
-            apt-get -qq update
-            apt-get -q install -y git openssh-client curl ca-certificates make tar gzip
+            sudo apt-get -qq update
+            sudo apt-get -q install -y git openssh-client curl ca-certificates make tar gzip
```
Comment (DvirDukhan) [inline .circleci/config.yml:41]: why sudo here?

File: .circleci/config.yml [hunk 2]
```diff
@@ -29,30 +37,30 @@ commands:
...
+#      - run:
+#          name: Relocate docker overlay2 dir
+#          command: |
+#            sudo systemctl stop docker
+#            sudo mkdir -p /var2/lib/docker
+#            sudo mv /var/lib/docker/overlay2 /var2/lib/docker
+#            sudo mkdir /var/lib/docker/overlay2
+#            sudo mount --bind /var2/lib/docker/overlay2 /var/lib/docker/overlay2
+#            sudo systemctl start docker
+#
```
Comment (DvirDukhan) [inline .circleci/config.yml:57]: you need it for docker GPU builds, otherwise you will run out of docker storage

File: .circleci/config.yml [hunk 3]
```diff
@@ -191,18 +199,17 @@ jobs:
           platform: debian
 
   platforms-build:
-    machine:
-      enabled: true
-      docker_layer_caching: true
-      resource_class: gpu.nvidia.small
-      image: ubuntu-1604-cuda-11.1:202012-01
+    docker:
+      - image: cimg/base:2020.01
```
Comment (DvirDukhan) [inline .circleci/config.yml:203]: what is this image?

File: .circleci/config.yml [hunk 4]
```diff
@@ -191,18 +199,17 @@ jobs:
...
     steps:
+      - setup-executor
       - platforms-build-steps
 
   coverage:
     docker:
       - image: redisfab/rmbuilder:6.2.1-x64-buster
     steps:
+      - abort_for_docs
```
Comment (DvirDukhan) [inline .circleci/config.yml:212]: duplicate

File: .circleci/config.yml [hunk 5]
```diff
@@ -434,17 +481,15 @@ on-integ-branch: &on-integ-branch
       only:
         - master
         - /^\d+\.\d+.*$/
-        - /^feature-.*$/
     tags:
       ignore: /.*/
 
-not-on-integ-branch: &not-on-integ-branch
+on-wip-branches: &on-wip-branches
```
Comment (DvirDukhan) [inline .circleci/config.yml:487]: maybe `dev_branches`

File: .circleci/config.yml [hunk 6]
```diff
@@ -514,15 +549,8 @@ workflows:
           <<: *on-integ-branch
           <<: *after-linter
       - build-and-test-gpu:
-          <<: *on-any-branch
+          <<: *on-integ-branch
```
Comment (DvirDukhan) [inline .circleci/config.yml:552]: we haven't decided on the design doc where this should be
let's iterate it

Comment (chayim) [reply .circleci/config.yml:552]: We'll move this to a github action similar to valgrind - but keep it this way for now, until that is ready.

File: .circleci/config.yml (hunk 1 above)
Comment (chayim) [reply .circleci/config.yml:41]: b/c the need to install packages.

File: .circleci/config.yml (hunk 3 above)
Comment (chayim) [reply .circleci/config.yml:203]: circleci base image.

File: .circleci/config.yml [hunk 7]
```diff
@@ -121,8 +129,8 @@ commands:
   platforms-build-steps:
     steps:
       - abort_for_docs
+      - setup-executor
       - checkout-all
-      - relocate-docker-storage
```
Comment (DvirDukhan) [inline .circleci/config.yml:125]: you need this one back, no?

**Review by DvirDukhan — APPROVED** [review]

**Pull request merged**